design/slot-reservations.md

```diff
@@ -0,0 +1,225 @@
+# Slot reservations
+
+Competition between hosts to fill slots has some advantages, such as providing
+an incentive for hosts to become proficient in downloading content and
+generating proofs. It also has some drawbacks, for instance it can lead to
+network inefficiencies because multiple hosts do the work of downloading and
+proving, while only one host is rewarded for it. These inefficiencies lead to
+higher costs for hosts, which leads to an overall increase in the price of
+storage on the network. It can also lead to clients inadvertently inviting too
+much network traffic to themselves. Should they for instance post a very
+lucrative storage request, then this invites a lot of hosts to start downloading
+the content from the client simultaneously, not unlike a DDOS attack.
+
+Slot reservations are a means to avoid these inefficiencies by only allowing SPs
+who have secured a slot reservation to fill the slot. Furthermore, slots can
+only be reserved by eligible SPs, governed by a window of eligible addresses
+that starts small and grows larger over time, eventually encompassing the entire
+address space on the network.
+
+## Proposed solution: slot reservations
+
+Before downloading the content associated with a slot, a limited number of hosts
+can reserve the slot. Only hosts that have reserved the slot can fill the slot.
+After the host downloads the content and calculates a proof, it can move the
+slot from its reserved state into the filled state by providing collateral and
+the storage proof. Then it begins to periodically provide storage proofs and
+accrue payments for the slot.
+
+```
+         reserve         proof & collateral
+            |                  |
+            v                  v
+            ---------------------------------------------
+     slot:  |/ / / / / / / / / |/////////////////////////
+            ---------------------------------------------
+            |                  |
+            v                  v
+          slot                slot
+         reserved            filled
+
+
+            ---------------- time ---------------->
+```
+
+There is an initial race for eligible SPs who are first to secure a reservation, then a
```

Comment:
> There is an initial race for eligible SPs who are first to secure a reservation

This race is not really related to our network as it is "settled" on blockchain. So IMHO not really an issue.

Reply:
True, there is not an impact on the Codex network, but it describes chronologically how an SP would fill a slot once eligible, no?

Reply:
Agree, but maybe I would only add a comment that first race is on blockchain, while second race is already on Codex network...

Reply:
The second race involves downloading data from the network (likely only those with a reservation would download), but it is still a race amongst reserved SPs to fill the slot on chain. I actually think it's an important point to describe that there are races, as this was a question that someone had when we initially presented the idea at the offsite, and perhaps it may clarify how it works for other readers.